### PR TITLE
add backend and update_state kwargs to show

### PR DESCRIPTION
**Pull request by bjarthur**
as [discussed](https://discourse.julialang.org/t/correct-way-to-save-makie-plot-to-iobuffer/87793) `show` is the way to "save" a figure to an IOBuffer.  but if the figure was made with say WGLMakie and you want to save as say PDF then it's a bit cumbersome because you have to `CairoMakie.activate!` and then `WGLMakie.activate!` after `show`ing.  this PR make it more concise by adding a `backend` kwarg to `show`, just like `save` has one already.  so one can simply `show(...; backend=CairoMakie)` instead of activating and deactivating it.
 
## Type of change

Delete options that do not apply:

- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] Added an entry in CHANGELOG.md (for new features and breaking changes)
- [x] Added unit tests for new algorithms, conversion methods, etc.

### CODE REVIEW

**Comment by bjarthur** [comment]
do any other `show` methods defined by Makie need to have a backend kwarg added as well?

also, i notice that `show` is not tested at all.  should i bother to add one?

**Comment by SimonDanisch** [comment]
Thank you, seems reasonable!

> should i bother to add one?

Yeah why not :)

**Comment by bjarthur** [comment]
all i can think to test is for the presence of a `backend` kwarg to `show`.

ideally i'd like to test that specifying a different backend works, but that would require putting the test in one of the backend repos and adding a dependency to one of the other backends.  you don't want to go that far, do you?

**Comment by bjarthur** [comment]
ready for review

**Comment by bjarthur** [comment]
the failed test on WGLMakie  for julia 1.6 is a memory leak.  is that related to this PR?

**Comment by bjarthur** [comment]
just added another kwarg to make reseting the limits optional when `show`'ing.

File: src/display.jl [hunk 1]
```diff
@@ -244,17 +244,16 @@ function Base.show(io::IO, m::MIME"text/markdown", fig::FigureLike)
     throw(MethodError(show, io, m, fig))
 end
 
-function Base.show(io::IO, m::MIME, figlike::FigureLike)
+function Base.show(io::IO, m::MIME, figlike::FigureLike; backend = current_backend(), update_state=true)
```
Comment (ffreyer) [inline src/display.jl:247]: I think just `update` would be better since that's what we use in `display()`

Comment (bjarthur) [reply src/display.jl]: oh, thanks!  i didn't notice that.  changed in a new commit.

**Pull request merged**